pykokkos/core/parsers/parser.py

```diff
@@ -123,19 +124,165 @@ def get_entities(self, style: PyKokkosStyles) -> Dict[str, PyKokkosEntity]:
 
         for i, node in enumerate(self.tree.body):
             if check_entity(node, self.pk_import):
-                start: int = node.lineno - 1
 
+                start: int = node.lineno - 1
                 try:
                     stop: int = self.tree.body[i + 1].lineno - 1
                 except IndexError:
                     stop = len(self.lines)
-
+                
                 name: str = node.name
+
                 entity = PyKokkosEntity(style, cppast.DeclRefExpr(name), node, (self.lines[start:stop], start), self.path, self.pk_import)
                 entities[name] = entity
 
         return entities
 
+
+    def fix_types(self, entity: PyKokkosEntity, updated_types: UpdatedTypes) -> ast.AST:
+
+        '''
+        updatedTypes: object that contains info about inferred types
+
+        This method will walk the ast for a workunit and add any mission annotations
+        This method will also invoke fix_viewlayouts that will add missing decorators for user specified layouts
+        '''
+        
+        check_entity: Callable[[ast.stmt], bool]
+        style: PyKokkosStyles = entity.style
+
+        # only supports standalone workunits, return the entity AST as it is
+        if style is not PyKokkosStyles.workunit or updated_types == None:
+            return entity.AST
```

Comment:
In what cases is this condition expected to be met? Are we calling this function from somewhere where the style might not be a workunit or `updated_types` might be None? I'd rather we have an assert statement here